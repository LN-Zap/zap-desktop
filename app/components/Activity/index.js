import Activity from './Activity'

export default Activity
