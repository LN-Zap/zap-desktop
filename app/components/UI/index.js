export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundTertiary from './BackgroundTertiary'
export BackgroundSecondary from './BackgroundSecondary'
export Bar from './Bar'
export Button from './Button'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export DataRow from './DataRow'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export Message from './Message'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export MainContent from './MainContent'
export Menu from './Menu'
export MenuItem from './MenuItem'
export MenuItemGroup from './MenuItemGroup'
export Modal from './Modal'
export Notification from './Notification'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export Page from './Page'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Truncate from './Truncate'
export Value from './Value'
export Wizard from './Wizard'
