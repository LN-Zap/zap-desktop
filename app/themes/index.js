import dark from './dark'
import light from './light'

export { dark, light }
