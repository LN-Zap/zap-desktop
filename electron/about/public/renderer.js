'use strict'
Object.defineProperty(exports, '__esModule', { value: true })

window.ipcRenderer.on('about-window:info', (_, info) => {
  const app_name = info.product_name
  const content = info.use_inner_html ? 'innerHTML' : 'innerText'
  document.title = `About ${app_name}`
  const title_elem = document.querySelector('.title')
  title_elem.innerText = `${app_name}`

  const copyright_elem = document.querySelector('.copyright')
  if (info.copyright) {
    copyright_elem[content] = info.copyright
  } else if (info.license) {
    copyright_elem[content] = `Distributed under ${info.license} license.`
  }
  const icon_elem = document.getElementById('app-icon')
  icon_elem.src = info.icon_path
  if (info.description) {
    const desc_elem = document.querySelector('.description')
    desc_elem[content] = info.description
  }

  if (info.css_path) {
    const css_paths = Array.isArray(info.css_path) ? info.css_path : [info.css_path]
    for (const css_path of css_paths) {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = css_path
      document.head.appendChild(link)
    }
  }

  if (info.use_version_info) {
    const versions = document.querySelector('.versions')
    const vs = process.versions
    for (const name of ['electron', 'chrome', 'node', 'v8']) {
      const tr = document.createElement('tr')
      const name_td = document.createElement('td')
      name_td.innerText = name
      tr.appendChild(name_td)
      const version_td = document.createElement('td')
      version_td.innerText = ' : ' + vs[name]
      tr.appendChild(version_td)
      versions.appendChild(tr)
    }
  }
})
//# sourceMappingURL=renderer.js.map
