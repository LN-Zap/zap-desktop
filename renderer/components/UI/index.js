export ActionBar from './ActionBar'
export ActionButton from './ActionButton'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export { BasicInput } from './Input'
export { BasicRange } from './Range'
export { BasicTextArea } from './TextArea'
export { BasicCheckbox } from './Checkbox'
export { BasicSelect } from './Select'
export Button from './Button'
export Card from './Card'
export Checkbox from './Checkbox'
export CloseButton from './CloseButton'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export DialogOverlay from './DialogOverlay'
export Donut from './Donut'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export LndConnectionStringInput from './LndConnectionStringInput'
export LndConnectionStringEditor from './LndConnectionStringEditor'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export { ModalOverlayStyles } from './Modal'
export NodePubkeyInput from './NodePubkeyInput'
export Notification from './Notification'
export OpenDialogButton from './OpenDialogButton'
export OpenDialogInput from './OpenDialogInput'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export ProgressBar from './ProgressBar'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioCard from './RadioCard'
export RadioGroup from './RadioGroup'
export Range from './Range'
export RowLabel from './RowLabel'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Tab from './Tab'
export Tabs from './Tabs'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Tooltip from './Tooltip'
export TransactionFeeInput from './TransactionFeeInput'
export Value from './Value'
export Wizard from './Wizard'
