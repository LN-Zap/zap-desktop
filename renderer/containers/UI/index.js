export CurrencyFieldGroup from './CurrencyFieldGroup'
export CryptoSelector from './CryptoSelector'
export CryptoValue from './CryptoValue'
export FiatSelector from './FiatSelector'
export FiatValue from './FiatValue'
