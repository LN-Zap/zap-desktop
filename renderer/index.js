import React from 'react'

import jstz from 'jstimezonedetect'
import ReactDOM from 'react-dom'
import { Provider } from 'react-intl-redux'

import { getDefaultLocale } from '@zap/i18n'
import translations from '@zap/i18n/translation'

import Root from './containers/Root'
import { configureStore, history } from './store/configureStore'

// Default the locale to English.
const defaultLocale = getDefaultLocale()

// Initialise the intl store with data from the users current locale.
const initialState = {
  intl: {
    locale: defaultLocale,
    messages: translations[defaultLocale],
    timeZone: jstz.determine().name(),
  },
}

// Set up the redux store.
const store = configureStore(initialState)
const MOUNT_NODE = document.getElementById('root')

const render = Component => {
  ReactDOM.render(
    <Provider store={store}>
      <Component history={history} />
    </Provider>,
    MOUNT_NODE
  )
}

render(Root)

if (module.hot) {
  module.hot.accept('./containers/Root', () => {
    render(Root)
  })
}
