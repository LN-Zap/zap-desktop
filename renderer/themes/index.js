import dark from './dark'
import light from './light'

export { dark }
export { light }

export default { dark, light }
