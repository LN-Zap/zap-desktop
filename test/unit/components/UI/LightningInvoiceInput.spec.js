import React from 'react'
import { Form } from 'informed'
import renderer from 'react-test-renderer'
import { dark } from 'themes'
import { ThemeProvider } from 'styled-components'
import { LightningInvoiceInput } from 'components/UI'
import { IntlProvider } from 'react-intl'

describe('component.UI.LightningInvoiceInput', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <IntlProvider locale="en">
          <ThemeProvider theme={dark}>
            <Form>
              <LightningInvoiceInput field="name" chain="bitcoin" network="mainnet" theme={dark} />
            </Form>
          </ThemeProvider>
        </IntlProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
