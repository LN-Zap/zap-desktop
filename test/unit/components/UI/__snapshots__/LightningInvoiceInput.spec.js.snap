// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.LightningInvoiceInput should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  padding: 16px;
  width: 100%;
  font-size: 13px;
  color: #ffffff;
  background-color: transparent;
  color: #ffffff;
  background-color: transparent;
  font-family: Roboto,system-ui,sans-serif;
  font-weight: 300;
  border: 1px solid;
  border: 1px solid;
  border-color: #959595;
  border-radius: 5px;
  outline: none;
}

.c1:not([readOnly]):not([disabled]):focus {
  border: 1px solid #fd9800;
}

<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="c0"
  >
    <textarea
      className="c1"
      color="primaryText"
      fontFamily="sans"
      fontSize="m"
      fontWeight="light"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      opacity={null}
      placeholder="Paste a Lightning Payment Request or Bitcoin Address here"
      required={false}
      rows={5}
      spellCheck="false"
      value=""
      width={1}
    />
    <div
      className="c2"
    />
  </div>
</form>
`;
